File: python/promethean/pseudo_random.py
import random

class PseudoRandom:
    def __init__(self, in_seed: int):
        self._random = random.Random(in_seed)

    def next(self, in_min: int, in_max: int) -> int:
        return self._random.randint(in_min, in_max)

    def next_odd(self, in_min: int, in_max: int) -> int:
        next_value: int = self.next(in_min, in_max)

        if next_value % 2 != 0:
            return next_value
        else:
            if next_value < in_max:
                return next_value + 1
            else:
                return next_value - 1

File: assemblyscript/promethean/assembly/pseudo_random.ts

export class PseudoRandom{
    next(in_min: i32, in_max: i32): i32{
        const v: f64 = Math.random();
        return <i32>(<f64>(in_max - in_min) * v + <f64>in_min + 0.5);
    }

    next_odd(in_min: i32, in_max: i32): i32{
        const next_value: i32 = this.next(in_min, in_max);

        if(next_value % 2 != 0){
            return next_value;
        }
        else{
            if(next_value < in_max){
                return next_value + 1;
            }
            else{
                return next_value - 1;
            }
        }
    }
}
